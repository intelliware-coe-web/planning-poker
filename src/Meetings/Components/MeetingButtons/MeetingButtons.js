import React from 'react';
import {Link} from 'react-router-dom';
import '../Meetings.css';
import {connect} from 'react-redux';
import {FindMeetings} from '../../Actions/MeetingsActions';

// const meetings = ['First', 'Second', 'Third', 'Fourth'];

function MeetingButtons({meetings}) {
    if (meetings.length === 0) {
        return null;
    }
    return meetings.map((meeting, index) => 
        <div key={index}>
                <Link to="/host">
                    <div className="uk-card uk-card-secondary uk-card-body meeting-card">
                        <h3 className="uk-card-title">{meeting}</h3>
                    </div>
                </Link>
            <br/>
        </div>
    );
}

function mapStateToProps() {
    return {
    }
 }
 function mapDispatchToProps(dispatch) {
    return {
        meetings: () => dispatch(FindMeetings())
    };
 }
 
 export default connect(mapStateToProps, mapDispatchToProps)(MeetingButtons)