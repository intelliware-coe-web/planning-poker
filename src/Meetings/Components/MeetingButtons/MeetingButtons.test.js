import MeetingButtons from "./MeetingButtons";

it('MeetingButtons renders correctly', () => {
    expect(MeetingButtons()).toMatchSnapshot(); 
});