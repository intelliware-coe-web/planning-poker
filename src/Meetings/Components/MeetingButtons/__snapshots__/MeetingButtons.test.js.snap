// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingButtons renders correctly 1`] = `
Array [
  <div>
    <Link
      to="/host"
    >
      <div
        className="uk-card uk-card-secondary uk-card-body meeting-card"
      >
        <h3
          className="uk-card-title"
        >
          First
        </h3>
      </div>
    </Link>
    <br />
  </div>,
  <div>
    <Link
      to="/host"
    >
      <div
        className="uk-card uk-card-secondary uk-card-body meeting-card"
      >
        <h3
          className="uk-card-title"
        >
          Second
        </h3>
      </div>
    </Link>
    <br />
  </div>,
  <div>
    <Link
      to="/host"
    >
      <div
        className="uk-card uk-card-secondary uk-card-body meeting-card"
      >
        <h3
          className="uk-card-title"
        >
          Third
        </h3>
      </div>
    </Link>
    <br />
  </div>,
  <div>
    <Link
      to="/host"
    >
      <div
        className="uk-card uk-card-secondary uk-card-body meeting-card"
      >
        <h3
          className="uk-card-title"
        >
          Fourth
        </h3>
      </div>
    </Link>
    <br />
  </div>,
]
`;
